hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml

```diff
@@ -0,0 +1,126 @@
+graphs: {
+}
+scriptEngines: {
+  gremlin-groovy: {
+    staticImports: [
+        org.opencypher.gremlin.process.traversal.CustomPredicates.*',
+        org.opencypher.gremlin.traversal.CustomFunctions.*
+    ],
+    plugins: {
+      org.apache.hugegraph.plugin.HugeGraphGremlinPlugin: {},
+      org.apache.tinkerpop.gremlin.server.jsr223.GremlinServerGremlinPlugin: {},
+      org.apache.tinkerpop.gremlin.jsr223.ImportGremlinPlugin: {
+        classImports: [
+          java.lang.Math,
+          org.apache.hugegraph.backend.id.IdGenerator,
+          org.apache.hugegraph.type.define.Directions,
+          org.apache.hugegraph.type.define.NodeRole,
+          org.apache.hugegraph.traversal.algorithm.CollectionPathsTraverser,
+          org.apache.hugegraph.traversal.algorithm.CountTraverser,
+          org.apache.hugegraph.traversal.algorithm.CustomizedCrosspointsTraverser,
+          org.apache.hugegraph.traversal.algorithm.CustomizePathsTraverser,
+          org.apache.hugegraph.traversal.algorithm.FusiformSimilarityTraverser,
+          org.apache.hugegraph.traversal.algorithm.HugeTraverser,
+          org.apache.hugegraph.traversal.algorithm.JaccardSimilarTraverser,
+          org.apache.hugegraph.traversal.algorithm.KneighborTraverser,
+          org.apache.hugegraph.traversal.algorithm.KoutTraverser,
+          org.apache.hugegraph.traversal.algorithm.MultiNodeShortestPathTraverser,
+          org.apache.hugegraph.traversal.algorithm.NeighborRankTraverser,
+          org.apache.hugegraph.traversal.algorithm.PathsTraverser,
+          org.apache.hugegraph.traversal.algorithm.PersonalRankTraverser,
+          org.apache.hugegraph.traversal.algorithm.SameNeighborTraverser,
+          org.apache.hugegraph.traversal.algorithm.ShortestPathTraverser,
+          org.apache.hugegraph.traversal.algorithm.SingleSourceShortestPathTraverser,
+          org.apache.hugegraph.traversal.algorithm.SubGraphTraverser,
+          org.apache.hugegraph.traversal.algorithm.TemplatePathsTraverser,
+          org.apache.hugegraph.traversal.algorithm.steps.EdgeStep,
+          org.apache.hugegraph.traversal.algorithm.steps.RepeatEdgeStep,
+          org.apache.hugegraph.traversal.algorithm.steps.WeightedEdgeStep,
+          org.apache.hugegraph.traversal.optimize.ConditionP,
+          org.apache.hugegraph.traversal.optimize.Text,
+          org.apache.hugegraph.traversal.optimize.TraversalUtil,
+          org.apache.hugegraph.util.DateUtil,
+          org.opencypher.gremlin.traversal.CustomFunctions,
+          org.opencypher.gremlin.traversal.CustomPredicate
+        ],
+        methodImports: [
+            java.lang.Math#*,
+            org.opencypher.gremlin.traversal.CustomPredicate#*,
+            org.opencypher.gremlin.traversal.CustomFunctions#*
+        ]
+      },
+      org.apache.tinkerpop.gremlin.jsr223.ScriptFileGremlinPlugin: {
+        files: [scripts/example-preload.groovy]
+      }
```

Comment:
if we only modify one line, consider use `sed` before load the file? 

so as the `pre-load` groovy file (could show the differ)

Reply:
> if we only modify one line, consider use `sed` before load the file?
> 
> so as the `pre-load` groovy file (could show the differ)
> 
> and @VGalaxies any suggestion for it to preload some data?

The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.

**If the content of this script can be guaranteed to remain unchanged**, deleting the fixed lines in `example.groovy` using `sed` can achieve the expected effect (keep `gremlin-server.yaml` unchanged).

Reply:
> > if we only modify one line, consider use `sed` before load the file?
> > so as the `pre-load` groovy file (could show the differ)
> > and @VGalaxies any suggestion for it to preload some data?
> 
> The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.
> 
> So, deleting the corresponding lines in `example.groovy` using `sed` can achieve the expected effect (no need to introduce `gremlin-server-preload.yaml` and `example-preload.groovy`).
> 
> ```shell
> sed -i -e '/registerRocksDB/d; /serverStarted/d' example.groovy
> ```

Thank you! And I have another question. 
If the users want to test the algorithm like PageRank, they need a demo dataset like movieLen(ml-latest-small) with more vertices and edges. 
Should we provide such feature? Or just preload these data by groovy in current version. And can we preload the file data like [hugegraph-loader](https://hugegraph.apache.org/docs/quickstart/hugegraph-loader/#44-command-to-import) in server or hubble?

Reply:
> > > if we only modify one line, consider use `sed` before load the file?
> > > so as the `pre-load` groovy file (could show the differ)
> > > and @VGalaxies any suggestion for it to preload some data?
> > 
> > 
> > The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.
> > So, deleting the corresponding lines in `example.groovy` using `sed` can achieve the expected effect (no need to introduce `gremlin-server-preload.yaml` and `example-preload.groovy`).
> > ```shell
> > sed -i -e '/registerRocksDB/d; /serverStarted/d' example.groovy
> > ```
> 
> Thank you! And I have another question. If the users want to test the algorithm like PageRank, they need a demo dataset like movieLen(ml-latest-small) with more vertices and edges. Should we provide such feature? Or just preload these data by groovy in current version. And can we preload the file data like [hugegraph-loader](https://hugegraph.apache.org/docs/quickstart/hugegraph-loader/#44-command-to-import) in server or hubble?

This is just my personal opinion, but I think it's enough to create a basic graph by preloading this data through the groovy script when server starting. Users can then use Loader or Hubble to import the data they need (like movieLen) 🤔

Reply:
> > > > if we only modify one line, consider use `sed` before load the file?
> > > > so as the `pre-load` groovy file (could show the differ)
> > > > and @VGalaxies any suggestion for it to preload some data?
> > > 
> > > 
> > > The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.
> > > So, deleting the corresponding lines in `example.groovy` using `sed` can achieve the expected effect (no need to introduce `gremlin-server-preload.yaml` and `example-preload.groovy`).
> > > ```shell
> > > sed -i -e '/registerRocksDB/d; /serverStarted/d' example.groovy
> > > ```
> > 
> > 
> > Thank you! And I have another question. If the users want to test the algorithm like PageRank, they need a demo dataset like movieLen(ml-latest-small) with more vertices and edges. Should we provide such feature? Or just preload these data by groovy in current version. And can we preload the file data like [hugegraph-loader](https://hugegraph.apache.org/docs/quickstart/hugegraph-loader/#44-command-to-import) in server or hubble?
> 
> This is just my personal opinion, but I think it's enough to create a basic graph by preloading this data through the groovy script when server starting. Users can then use Loader or Hubble to import the data they need (like movieLen) 🤔

All right, thx for your opinion.